src/v/bytes/iobuf.h

```diff
@@ -238,8 +240,7 @@ inline void iobuf::create_new_fragment(size_t sz) {
     oncore_debug_verify(_verify_shard);
     auto chunk_max = std::max(sz, last_allocation_size());
     auto asz = details::io_allocation_size::next_allocation_size(chunk_max);
-    auto f = new fragment(ss::temporary_buffer<char>(asz), fragment::empty{});
-    append_take_ownership(f);
+    append(std::make_unique<fragment>(asz));
```

Comment:
Why are fragments heap allocated?

Reply:
because they are on an intrusive list

Reply:
Ahh yes, of course. I remember the API now.